Document minimal config files

texastoland opened the pull request:
#### 3 New "Tips"

- Minimal configs vs starting from defaults
- `std path add` to simplify editing the system path
- ~~`set-env` from nu_scripts to simplify editing nested configs~~

#### Additional Amendments:

- Explained more concretely why `env.nu` vs `config.nu`
- Explained why `upsert` is necessary
- Reworded the editors section and removed parts about old Nu versions
- Brief note about `config.nu` not loading non-interactively
- Simplify `nuopen` command to alias
- Reformatted `$env.PATH` snippet more conventionally
- Changed 2 heading levels to improve TOC
- Extra links in general

fdncred commented:
I'm not sure I'd recommend set-env in our official docs when it's not officially a part of nushell yet.

texastoland commented:
Yeah that's why I didn't imply it's standard in  any way. Just makes a day 1 tricky task easier. I have no reservation about removing that block regardless!

fdncred commented:
I'm fine with `path add` since it's in the stdlib, just not `set-env` since it's not there yet.

texastoland commented:
My bad fixed 🙉

texastoland commented:
@fdncred I added 2 infra lines:

```diff
+const [ciUser, ciRepo] = process.env.GITHUB_REPOSITORY?.split('/') ?? []

 export default defineUserConfig({
+  base: ciRepo && ciUser !== 'nushell' ? `/${ciRepo}/` : '/',
```

1. Get the user and repo from the GitHub action (or undefined)
2. If there's a fork it changes the base URL otherwise (nushell org and local) it uses the default.

fdncred commented:
>  to enable my live preview in the description

I'd guess this isn't a good idea because it's specific to your environment and no one else's.

texastoland commented:
You can try it in your own fork by enabling actions there 💯

fdncred commented:
Are you saying this is a github thing or something specific to your env vars?

texastoland commented:
Added comments for future contributors:

```diff
+// default env from the deploy GitHub action
+// e.g. ciUser = nushell and ciRepo = nushell.github.io
+// both default to undefined if the env is undefined
 const [ciUser, ciRepo] = process.env.GITHUB_REPOSITORY?.split('/') ?? []

 export default defineUserConfig({
+  // set the base URL to ciRepo dir if it's a fork
+  // keep the default root if not
   base: ciRepo && ciUser !== 'nushell' ? `/${ciRepo}/` : '/',
```

Also in `CONTRIBUTING.md`:

```md
## Enabling a Preview URL From Your Fork

Just enable GitHub actions in your repo settings. That's all! It will start deploying the next time you push to `main`.
```

fdncred commented:
ok, thanks
